pkgs/servers/samba/4.x.nix

```diff
@@ -29,6 +31,20 @@ stdenv.mkDerivation rec {
     sha256 = "0vkxqp3wh7bpn1fd45lznmrpn2ma1fq75yq28vi08rggr07y7v8y";
   };
 
+  srcSignature = fetchurl {
+    url = "mirror://samba/pub/samba/stable/${name}.tar.asc";
+    sha256 = "0wpcbwbs1bj1y0amhn0z29v55f2hhmzc5p8n7sbwg9kaf0hc5mz5";
+  };
+  signatureUncompressed = true;
+
+  signaturePublicKey = fetchpgpkey {
+    url = https://download.samba.org/pub/samba/samba-pubkey.asc;
+    sha256 = "1fndhq0c34va34z137gvsl9gpwjv30b06makfx8cq5vrmgiax1x1";
+    fingerprint = "52FBC0B86D954B0843324CDC6F33915B6568B7EA";
+  };
+
+  nativeBuildInputs = [ verifySignatureHook ];
```

Comment:
Does this work? We aren't downloading binaries here so I have no idea where the signature would be coming from. This should be not needed at all for source based builds.

Reply:
It works by convention: The hook assumes that  `srcSignature` contains the signature file, and `signaturePublicKey` the corresponding key. If they're present, `srcSignature` is checked against `src`.  Of course you have to `fetchurl` the signature file and `fetchpgpkey` the key for the hook to work.

Reply:
And yes, it can be used for source base builds if upstream provides a signature for the source tarball. It obviously doesn't work with `fetchFromGitHub`, but that would be a case for verifying signed git commits.

Reply:
Ok did not realize that on tarball signatures! I wonder if we could put the source-based part of the hook in fetchurl then? Just have an optional attribute that is verified for us. This would also make it possible to support multiple-src derivations.